### ISSUE TITLE
New rule: Specify dot location (before/after line break)

### ISSUE THREAD

**Issue by feross** [eslint/eslint#1884]
I would like a rule to disallow:

```js
someObject.
  someMethod()
```

And allow:

```js
someObject
  .someMethod()
```

Or vice-versa.

Thoughts on adding such a rule?

This is related to https://github.com/eslint/eslint/issues/1405, but would be a different rule since the dot is not considered an operator, according to @nzakas.

**Comment by lo1tuma** [comment]
Sounds like a good idea to me :+1:.

**Comment by nzakas** [comment]
Yeah, love this idea.

**Comment by gcochard** [comment]
I propose naming the rule `dot-line-breaks` after the similarly named [operator-line-breaks](https://github.com/ilovejary/eslint/commit/4b9ebed82edacb45b3add81be0be2b2962b065ab).

Config should be a string with value "before" or "after"

What does everyone think?

Should we simply ignore single lines for this rule and always allow them? I can't really imagine someone wanting **every** dot to be preceded/followed by a line break.

If there is demand for this, the extra config `{"allowSingleLine":true}` could be used.

I can handle creating this rule this weekend.

**Comment by lo1tuma** [comment]
Maybe it would be easier to specify if you want the to dot to be on the same line as the `object` of the `MemberEpression` or on the same line as the `property` of the `MemberExpression`. If everything is on the same line it still works fine and you don't have to define any exceptions.

**Comment by gcochard** [comment]
That makes sense. Can you think of a name for the rule that fits this concept better? I am at a loss.

Maybe `dot-location`, with the config being "withObject" or "withProperty"?

**Comment by nzakas** [comment]
I'd say just "object" or "property".

**Comment by doberkofler** [comment]
Great idea.

**Comment by gcochard** [comment]
An update on this. I didn't get a chance to work on it last weekend. I hope to this weekend.

**Comment by gcochard** [comment]
Should have a PR ready tonight.

**Comment by gcochard** [comment]
One last thing, would it be desirable to have this enforce a dot attached to the object or property, when there is whitespace? For example `object. property` vs `object .property`? I think the name of the rule implies that the above *should* be the behavior; e.g. attach the dot to the object or the property.

**Comment by nzakas** [comment]
I'm not sure - that seems like it could be a separate rule.

**Comment by IanVS** [comment]
@gcochard I'd like to use a rule like this.  Are you still working on it?

**Comment by gcochard** [comment]
@IanVS Yes, it has fallen down in priority for me but I have some of it implemented. I can get it finished this week.

**Comment by gcochard** [comment]
Finally an update on this.

I don't know why, but my machine freezes every time I try to run `npm test`. I haven't had time to debug this, but I'll have an implementation for this rule completed shortly. I'll rely on travis to validate my changes.
